spring-core/src/test/java/org/springframework/util/unit/DataSizeTests.java

```diff
@@ -200,12 +200,32 @@ void isNegativeWithNegative() {
 
 	@Test
 	void toStringUsesBytes() {
-		assertThat(DataSize.ofKilobytes(1).toString()).isEqualTo("1024B");
+		assertThat(DataSize.ofBytes(1023).toString()).isEqualTo("1023B");
+	}
+
+	@Test
+	void toStringUsesKilobytes() {
+		assertThat(DataSize.ofKilobytes(1).toString()).isEqualTo("1KB");
```

Comment:
Rather than only testing the happy path, I'd like to see some tests where we can't fallback to the optimisation.